testsuites/store/src/main/java/org/eclipse/rdf4j/sail/SailConcurrencyTest.java

```diff
@@ -127,7 +127,7 @@ public void run() {
 						conn.addStatement(subject, RDFS.COMMENT, vf.createLiteral("ci" + i), context);
 					}
 					if (rollback) {
-						otherTxnCommitted.await();
+						otherTxnCommitted.await(); // deadlocks here. Waiting for another transaction while holding the exclusive write lock!
 						conn.rollback();
```

Comment:
The above await() waits for another transaction to finish while holding the exclusive write lock. The rollback() will release the lock, but the await() never returns.